datalad/support/gitrepo.py

```diff
@@ -70,7 +70,32 @@
 # log Exceptions from git commands.
 
 
-# TODO: ignore leading and/or trailing underscore to allow for python-reserved words
+# manipulate GitPython's command call wrapper to provide persistent git options,
+# while attribute `_git_options` keeps its functionality (and will be reset by
+# GitPython after each git call):
+class PatchedGit(gitpy.Git):
+
+    __slots__ = gitpy.Git.__slots__ + ('_git_persistent_options',
+                                       '_git_temp_options')
+
+    def __init__(self, *args, **kwargs):
+        super(PatchedGit, self).__init__(*args, **kwargs)
+        self._git_persistent_options = []
+        self._git_temp_options = ()
+
+    def _git_options_get(self):
+        return self._git_persistent_options + [i for i in self._git_temp_options]
+
+    def _git_options_set(self, val):
+        self._git_temp_options = val
+
+    _git_options = property(_git_options_get, _git_options_set)
```

Comment:
Sadly, it doesn't work that way, because this leads to modifying `_git_options` only if there were any git options passed at all. Whenever the original class (`Git`) creates a call without any git options passed, `__call__` will not be called and therefore `_git_persistent_options` are ignored. We need  `_git_options` to return something different, not being set different.